### PR TITLE
Add script to generate ETH wallpapers

### PR DESCRIPTION
**Pull request by adtac**

### CODE REVIEW

File: eth-wallpaper-gen/main.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from shapely.geometry.polygon import Polygon
+
+def scale(ms, width, height):
+    return [(x*width, y*height) for (x, y) in ms]
+
+magics = {
+    "top-left":     [(0.498, 0.183), (0.393, 0.445), (0.498, 0.538), (0.498, 0.183)],
+    "top-right":    [(0.498, 0.183), (0.604, 0.445), (0.498, 0.538), (0.498, 0.183)],
+    "bottom-left":  [(0.392, 0.472), (0.498, 0.567), (0.498, 0.697), (0.392, 0.472)],
+    "bottom-right": [(0.605, 0.472), (0.498, 0.567), (0.498, 0.697), (0.605, 0.472)],
```
Comment (adtac) [inline eth-wallpaper-gen/main.py]: If you're wondering what these magic numbers are, they are just polygon vertice points (relative to image dimensions).

File: eth-wallpaper-gen/main.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import sys
```
Comment (mbeacom) [inline eth-wallpaper-gen/main.py]: Can you modify this to be an executable python script?

File: eth-wallpaper-gen/main.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+mirrors = {
+    "top-left":     "bottom-right",
+    "top-right":    "bottom-left",
+    "bottom-left":  "top-right",
+    "bottom-right": "top-left",
+}
+
+im = Image.open(sys.argv[1])
```
Comment (mbeacom) [inline eth-wallpaper-gen/main.py]: We probably want to organize all of this logic into methods and implement a `main` method that calls the specific functions.

**Comment by owocki** [comment]
left some comments back on the issue itself

File: eth-wallpaper-gen/main.py (hunk 3 above)
Comment (adtac) [reply eth-wallpaper-gen/main.py]: Done :+1:

File: eth-wallpaper-gen/main.py (hunk 2 above)
Comment (adtac) [reply eth-wallpaper-gen/main.py]: Done :+1:

**Comment by adtac** [comment]
@owocki @mbeacom 

 - You can now run the script as an executable from the command line. The sytax is `./main.py input.jpg output.jpg [ETH_SCALE]`.

 - The third arg (`ETH_SCALE`) defines how large/small the ETH logo will be. A value of 1 means it'll be sized the same way as the foundation wallpapers.

 - Moved logical components into separate functions and created a `main` method to handle the processing (check out the code).

 - Applying a 30% opacity filter on top of each diamond segment to make the logo look more prominent.

**Comment by owocki** [comment]
grand_tetons
output


awesome!

**Pull request merged**